### PR TITLE
Features/hand 

### PR DESCRIPTION
**Pull request by izzytilles**
Contains Features/deck as well. 
Edits Files:
BasicCard.java
Deck.java
EventCard.java
Hand.java

### CODE REVIEW

**Review by tonixsmm — APPROVED** [review]

File: src/main/java/edu/gonzaga/Deck.java [hunk 1]
```diff
@@ -1,40 +1,140 @@
...
-    public void shuffle() {
+    /**
+     * Draws the first card from the deck
+     * @return the first card from the deck
+     * 
+     * @author Tony
+     */
+    public Card drawCard() {
+        // if the draw pile is empty, shuffle the discard pile and put it on top of the draw pile
+        if (drawPile.isEmpty()) {
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/Deck.java:107]: If the draw pile is empty in Pandemic, the game ends and shouldn't reshuffle the discard pile on top

File: src/main/java/edu/gonzaga/Deck.java [hunk 2]
```diff
@@ -1,40 +1,140 @@
...
 
+    /**
+     * Draw the bottom card of the deck
+     * @return the bottom card of the deck
+     * 
+     * @author Tony
+     */
+    public Card drawBottomCard() {
+        // if the draw pile is empty, shuffle the discard pile and put it on top of the draw pile
+        if (drawPile.isEmpty()) {
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/Deck.java:122]: Same comment from above, if the drawPile is empty then it's a losing condition and the game ends

File: src/main/java/edu/gonzaga/Deck.java [hunk 3]
```diff
@@ -1,40 +1,140 @@
...
 
+     /**
+     * Draw the bottom card of the deck
+     * 
+     * @author Tony
+     */
     public void putShuffledDiscardPileOnTop() {
-
+        // Side note: Can we agree that the top of the card is the beginning of the list and the bottom is the end?
+        shufflePile(discardPile); // NOTE: do we want to shuffle the discard pile before putting it on top?
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/Deck.java:136]: The infection pile will use this a lot, it should be shuffled!

File: src/main/java/edu/gonzaga/EventCard.java [hunk 4]
```diff
@@ -3,6 +3,17 @@
...
+     * @param name the name of the event card
+     * @return an EventCard object
+     * 
+     * @author Tony
+     */
+    public EventCard(String name) {
+        this.name = name;
+    }
+    
     public void playEvent() {
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/EventCard.java:17]: If we aren't going to use the playEvent() function let's just remove it

File: src/main/java/edu/gonzaga/Hand.java [hunk 5]
```diff
@@ -1,32 +1,186 @@
...
 
+    /**
+     * DVC for a Hand object - sets the limit to 7, declares it a valid hand to start 
+     * 
+     * @author Izzy T
+     */
     public Hand() {
+        handLimit = 7; // 7 basic/event cards allowed in a Hand 
+        validHand = true;
+        numCardsToCure = 5; // 5 cards of a color to cure cities of that color
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/Hand.java:35]: Can you make another constructor that allows me to set the numCardsToCure to a different variable (for one of the specific roles)

File: src/main/java/edu/gonzaga/Hand.java [hunk 6]
```diff
@@ -1,32 +1,186 @@
...
 
+    /**
+     * Restricts number of cards in a Hand - if card # exceeds max num allowed, then it requests one card be discarded until doesHandFollowLimit() is true
+     * 
+     * @author Izzy T
+     */
+    public void makeHandValid(Scanner userInput) {
+        int targetVal; // index of card list to delete a card from 
+        while (!this.doesHandFollowLimit()) {
+            System.out.println("Your hand exceeds the maximum card amount of " + handLimit + ". What card would you like to get rid of? ");
```
Comment (IsaacSShade) [inline src/main/java/edu/gonzaga/Hand.java:169]: Can you add a //TODO: comment for adding functionality with the UI? We should probably also add a card to Trello

**Review by IsaacSShade — CHANGES_REQUESTED** [review]
I left comments throughout the code. I'm fine with merging but if you agree with them, can you make those changes / make Trello cards to represent issues that need to be worked on in the future?

File: src/main/java/edu/gonzaga/EventCard.java (hunk 4 above)
Comment (izzytilles) [reply src/main/java/edu/gonzaga/EventCard.java:17]: The event card class is in progress right now, this merge is primarily for the Hand and Deck classes

**Review by IsaacSShade — APPROVED** [review]
All good

**Pull request merged**